qa/rpc-tests/test_framework/bumessages.py

```diff
@@ -309,7 +309,7 @@ class msg_get_xthin(object):
 
     def __init__(self, inv=None, filter=None):
         self.inv = inv
-        self.filter = filter
+        self.filter = filter if filter != None else CBloomFilter()
```

Comment:
That would be bad, because objects created in the default params line in python are only created once per program run.  Check this out:

```
class foo:
    pass

def bar(z=foo()):
    print(z)

def Test():
    bar()
    bar()
```
>>> import pdb ; import test; test.Test()
<test.foo object at 0x7f5551509ac8>
<test.foo object at 0x7f5551509ac8>

Reply:
TIL 😱